### ISSUE TITLE
Inventory updates work incorrectly when cancelling an order

### ISSUE THREAD

**Issue by kieckhafer** [reactioncommerce/reaction#3678]
### Expected behavior
- If I cancel an order, and choose to restock, my available inventory for the product should increase by the number of items I've "cancelled". For example, if I started with 20 products, ordered 5, captured the payment, and then cancelled the order with restocking, I should have 20 available products.
- If I cancel an order, and choose not to restock, my available inventory for the product should equal the original quantity, minus the quantity of this order. For example, if I started with 20 products, ordered 5, captured the payment, and then cancelled the order without restocking, I should have 15 available products.

### Actual behavior
- If I cancel an order, and choose to restock, my available inventory for the product stays at the depleted amount. For example, if I started with 20 products, ordered 5, captured the payment (the inventory drops to 15), and then cancelled the order **with** restocking, I still have 15 in my inventory.

https://youtu.be/PJKnsPruzMw

- If I cancel an order, and choose not to restock, my available inventory for the product is taken away twice, instead of just once. For example, if I started with 20 products, ordered 5, captured the payment (the inventory drops to 15), and then cancelled the order without restocking, the inventory drops an additional 5, and I now have 10 in my inventory (instead of 15).

https://youtu.be/izMjluy9fEQ

### Steps to reproduce the behavior
1. Write down the inventory of a product, and then create an order with X amount of the product
2. As an admin, begin to process the order, and capture the payement
3. Revisit the product and see that inventory has dropped by X
4. Cancel the order, choose "without restocking"
5 Revisit the product and see that inventory has dropped by an additional X

---

1. Write down the inventory of a product, and then create an order with X amount of the product
2. As an admin, begin to process the order, and capture the payement
3. Revisit the product and see that inventory has dropped by X
4. Cancel the order, choose "with restocking"
5 Revisit the product and see that inventory is still at the initial inventory - X, instead of being at the original inventory number

### Versions

```
Node: 9.5.0
NPM: 5.6.0
Meteor Node: 8.9.3
Meteor NPM: 5.5.1
Reaction CLI: 0.27.0
Reaction: 1.8.0
Reaction branch: release-1.8.0
Docker: 18.01.0-ce
```

Browser: http://whatismybrowser.com/w/KE449ZE

### QA

QA by @kieckhafer + @cassytaylor

**Comment by impactmass** [comment]
Findings:
#1.  Within the cancelOrder method, the restock logic is like thus:
```
if (!returnToStock) {
  Products.update({
  ...
  }, {
    $inc: {
      inventoryQuantity: -item.quantity
    }
  });
}
```

**Comment by impactmass** [comment]
**Findings**:

Within the cancelOrder method, the restock logic is like this:
```
if (!returnToStock) {
  Products.update({
  ...
  }, {
    $inc: {
      inventoryQuantity: -item.quantity
    }
  });
}
```
which means that when when the admin decides NOT to restock, the product quantity still gets decreased. This is led to the problem of double decrease in quantity (first at order capture, then second at order cancellation). This was introduced in #2022 (and #3513).

The correct logic should be:
```
if (returnToStock) {
  Products.update({
  ...
  }, {
    $inc: {
      inventoryQuantity: +item.quantity
    }
  });
}
```
which will mean that when user decides to return to stock, we go ahead and increment the current inventory with the canceled item quantity.

That way, 
(a) if I cancel an order, and choose to restock, my available inventory for the product will increase by the number of items I've "cancelled".
(b) If I cancel an order, and choose NOT to restock, my available inventory for the product will stay as it was when the order was captured

Before going with this fix, I have to find out if there's some reason the orderCancel was setup this way, as all the tests were also written to follow that logic

**Comment by zenweasel** [comment]
@impactmass I feel that probably there wasn't a good reason besides not thinking it through.

**Comment by spencern** [comment]
Closed by #3776
